pylint/lint.py

```diff
@@ -863,15 +863,25 @@ def process_tokens(self, tokens):
                             self.add_message("file-ignored", line=start[0])
                             self._ignore_file = True
                             return
-                        # If we did not see a newline between the previous line and now,
-                        # we saw a backslash so treat the two lines as one.
+                            # If we did not see a newline between the previous line and now,
+                            # we saw a backslash so treat the two lines as one.
+                        l_start = start[0]
                         if not saw_newline:
-                            meth(msgid, "module", start[0] - 1)
-                        meth(msgid, "module", start[0])
-                    except exceptions.UnknownMessageError:
-                        self.add_message("bad-option-value", args=msgid, line=start[0])
-            else:
-                self.add_message("unrecognized-inline-option", args=opt, line=start[0])
+                            l_start -= 1
+                        try:
+                            meth(msgid, "module", l_start)
+                        except exceptions.UnknownMessageError:
+                            self.add_message(
+                                "bad-option-value", args=msgid, line=start[0]
+                            )
+            except (UnknownKeyword, UnsupportedAssignment) as err:
```

Comment:
Ah I see now where we need all these exceptions. It looks to me though that two should be enough, one to convey the error for a valid, but unrecognized option, and another for a completely invalid pragma.

Reply:
I agree with you @PCManticore . The changes have been done.